src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReportService;
+import java.util.Map;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String SEPARATOR_IN_LINE = ", ";
+    private static final StringBuilder report = new StringBuilder();
+
+    @Override
+    public String generateReport(Map<String, Integer> db) {
+        for (String key : db.keySet()) {
```

Comment:
Use entrySet() instead of keySet() to iterate over the map, to avoid calling get(key) which adds an extra lookup.